shannonlal/mcp-postman
# feat: Add example Express API with Postman tests

### Task List

- [x] You have included code examples or test cases demonstrating your changes
- [x] You have completed local testing with Postman collections

### Description

This PR adds an example Express API implementation with Postman tests to demonstrate the MCP Postman server's capabilities. The example includes:

- A TypeScript Express server with GET and POST endpoints for managing items
- Postman collection and environment files for testing the API
- Shared dependencies with the root project for better maintainability
- ESM support and proper TypeScript configuration

Current behavior: No example implementation exists.
Updated behavior: Developers can run and test a working example API using the provided code and Postman tests.

### Testing Notes

To test this PR:

1. Start the example server:
```bash
pnpm run example:dev
```

2. Run the Postman tests:
```bash
pnpm newman run example/postman/collection.json -e example/postman/environment.json
```

Expected behavior:
- Server starts on http://localhost:3000
- GET /api/items returns an array of items
- POST /api/items creates a new item
- All Postman tests pass

### Example Usage

```typescript
// Example API endpoints
app.get('/api/items', (_req: Request, res: Response) => {
  res.json(items);
});

app.post('/api/items', (req: Request, res: Response) => {
  const { name, description } = req.body as CreateItemRequest;
  const newItem: Item = {
    id: Date.now().toString(),
    name,
    description,
    createdAt: new Date().toISOString()
  };
  items.push(newItem);
  res.status(201).json(newItem);
});
```

### Test Results

```
Example API Tests

→ GET Items
  GET http://localhost:3000/api/items [200 OK]
  ✓ Status code is 200
  ✓ Response is an array
  ✓ Response time is acceptable

→ Create Item
  POST http://localhost:3000/api/items [201 Created]
  ✓ Status code is 201
  ✓ Response has correct structure
  ✓ Created item matches request

┌─────────────────────────┬──────────────────┬──────────────────┐
│              assertions │                6 │                0 │
└─────────────────────────┴──────────────────┴──────────────────┘
```

### Additional Context

This example implementation serves as a reference for developers to understand how to:
1. Set up a TypeScript Express server
2. Create Postman collections for API testing
3. Configure shared dependencies
4. Use ESM modules with TypeScript
